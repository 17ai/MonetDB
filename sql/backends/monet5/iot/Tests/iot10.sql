-- use accumulated aggregation
set schema iot;
set optimizer='iot_pipe';

create stream table stmp10 (t timestamp, sensor integer, val decimal(8,2)) ;
create table tmp_aggregate(tmp_total decimal(8,2), tmp_count decimal(8,2));
insert into tmp_aggregate values(0.0,0.0);

create procedure collector()
begin
	update tmp_aggregate
		set tmp_total = tmp_total + (select sum(val) from iot.stmp10),
			tmp_count = tmp_count + (select count(*) from iot.stmp10);
	delete from iot.stmp10;
end;

insert into stmp10 values('2005-09-23 12:34:26.736',1,12.34);
select * from stmp10;

call iot.query('iot','collector');

--select * from iot.baskets();
--select * from iot.queries();

call iot.resume();
call iot.wait(1);

select * from tmp_aggregate;

drop table stmp10;
call iot.stop();
