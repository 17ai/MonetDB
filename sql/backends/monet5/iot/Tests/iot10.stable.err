stderr of test 'iot10` in directory 'sql/backends/monet5/iot` itself:


# 23:37:15 >  
# 23:37:15 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=38558" "--set" "mapi_usock=/var/tmp/mtest-25798/.s.monetdb.38558" "--set" "monet_prompt=" "--forcemito" "--dbpath=/export/scratch1/mk/iot//Linux/var/MonetDB/mTests_sql_backends_monet5_iot"
# 23:37:15 >  

# builtin opt 	gdk_dbpath = /export/scratch1/mk/iot//Linux/var/monetdb5/dbfarm/demo
# builtin opt 	gdk_debug = 0
# builtin opt 	gdk_vmtrim = no
# builtin opt 	monet_prompt = >
# builtin opt 	monet_daemon = no
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 38558
# cmdline opt 	mapi_usock = /var/tmp/mtest-25798/.s.monetdb.38558
# cmdline opt 	monet_prompt = 
# cmdline opt 	gdk_dbpath = /export/scratch1/mk/iot//Linux/var/MonetDB/mTests_sql_backends_monet5_iot
# cmdline opt 	gdk_debug = 536870922

# 23:37:15 >  
# 23:37:15 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-25798" "--port=38558"
# 23:37:15 >  


# 23:37:15 >  
# 23:37:15 >  "Done."
# 23:37:15 >  

