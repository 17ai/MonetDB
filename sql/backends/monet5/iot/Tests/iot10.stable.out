stdout of test 'iot10` in directory 'sql/backends/monet5/iot` itself:


# 23:37:15 >  
# 23:37:15 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=38558" "--set" "mapi_usock=/var/tmp/mtest-25798/.s.monetdb.38558" "--set" "monet_prompt=" "--forcemito" "--dbpath=/export/scratch1/mk/iot//Linux/var/MonetDB/mTests_sql_backends_monet5_iot"
# 23:37:15 >  

# MonetDB 5 server v11.24.0
# This is an unreleased version
# Serving database 'mTests_sql_backends_monet5_iot', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs and 128bit integers dynamically linked
# Found 15.589 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2016 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://vienna.da.cwi.nl:38558/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-25798/.s.monetdb.38558
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
# MonetDB/iot loaded

Ready.

# 23:37:15 >  
# 23:37:15 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-25798" "--port=38558"
# 23:37:15 >  

#set schema iot;
#set optimizer='iot_pipe';
#create stream table stmp10 (t timestamp, sensor integer, val decimal(8,2)) ;
#create table tmp_aggregate(tmp_total decimal(8,2), tmp_count decimal(8,2));
#insert into tmp_aggregate values(0.0,0.0);
[ 1	]
#create procedure collector()
#begin
#	update tmp_aggregate
#		set tmp_total = tmp_total + (select sum(val) from iot.stmp10),
#			tmp_count = tmp_count + (select count(*) from iot.stmp10);
#	delete from iot.stmp10;
#end;
#insert into stmp10 values('2005-09-23 12:34:26.000',1,9.0);
[ 1	]
#insert into stmp10 values('2005-09-23 12:34:27.000',1,11.0);
[ 1	]
#insert into stmp10 values('2005-09-23 12:34:28.000',1,13.0);
[ 1	]
#insert into stmp10 values('2005-09-23 12:34:28.000',1,15.0);
[ 1	]
#select * from stmp10;
% iot.stmp10,	iot.stmp10,	iot.stmp10 # table_name
% t,	sensor,	val # name
% timestamp,	int,	decimal # type
% 26,	1,	10 # length
[ 2005-09-23 12:34:26.000000,	1,	9.00	]
[ 2005-09-23 12:34:27.000000,	1,	11.00	]
#select * from tmp_aggregate;
% iot.tmp_aggregate,	iot.tmp_aggregate # table_name
% tmp_total,	tmp_count # name
% decimal,	decimal # type
% 10,	10 # length
[ 48.00,	4.00	]
#select * from iot.errors();
% .,	. # table_name
% table,	error # name
% clob,	clob # type
% 0,	0 # length
#drop procedure collector;
#drop table stmp10;

# 23:37:15 >  
# 23:37:15 >  "Done."
# 23:37:15 >  

